destination/parse.go

```diff
@@ -197,15 +197,28 @@ func (d deviceFinder) filterDeviceList(wantedDevice Simulator) (Device, error) {
 		return Device{}, fmt.Errorf("runtime (%s) not found", runtimeID)
 	}
 
-	deviceTypeIdentifier, err := d.lookupDeviceTypeID(wantedDevice.Name)
-	if err != nil {
-		return Device{}, err
+	// As the name of the device matches the device type ('iPhone 11') for factory created devices, look up device by name.
+	// If the default name is required and already created will use that.
+	for _, device := range devices {
+		if device.Name == wantedDevice.Name {
+			if !device.IsAvailable {
+				return Device{}, fmt.Errorf("device (%s) with runtime OS (%s) is unavailable: %s", wantedDevice.Name, runtime.Version, device.AvailabilityError)
```

Comment:
Can't we continue with one of the fallback options instead of failing with an error?

Reply:
It is not worth it, this case only happens if a specific device is already created but this field contains an error (like: ``` "devices" : {
	    "com.apple.CoreSimulator.SimRuntime.watchOS-7-4" : [
	      {
	        "availabilityError" : "runtime profile not found",```)